src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlParameter.cs

```diff
@@ -1620,7 +1637,11 @@ internal SmiParameterMetaData GetMetadataForTypeInfo()
         internal SmiParameterMetaData MetaDataForSmi(out ParameterPeekAheadValue peekAhead)
         {
             peekAhead = null;
-            MetaType mt = ValidateTypeLengths(true /* Yukon or newer */ );
+            MetaType mt = ValidateTypeLengths(
+#if NETFRAMEWORK
```

Comment:
Yukon is SQL 2005 which isn't supported anymore. So everything we will ever touch from this library is yukon or later which is why netcore didn't contain this code anymore. I don't think keeping this code is useful.

Reply:
I will do it in separate PR after discussing internally, as this check is in many files not just here.